scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,153 @@
+    function Log([string]$logStatement, [string]$color)
+    {
+        if(-not ([string]::IsNullOrEmpty($color)))
+        {
+            Write-Host "$($(Get-Date).ToString()): $logStatement" -ForegroundColor $color
+        }
+        else
+        { 
+            Write-Host "$($(Get-Date).ToString()): $logStatement"
+        }
+    }
+
+    function GetAbsolutePath([string]$Path)
+    {
+        $Path = [System.IO.Path]::Combine(((pwd).Path), ($Path));
+        $Path = [System.IO.Path]::GetFullPath($Path);
+        return $Path;
+    }
+
+    function OutFileWithCreateFolders([string]$path, [string]$content){
+        $folder = [System.IO.Path]::GetDirectoryName($path)
+        If(!(test-path $folder))
+        {
+            & New-Item -ItemType Directory -Force -Path $folder > $null
+        }
+        Add-Content -Path $path -Value $content
+    }
+
+    function GetAllPackagesInGlobalPackagesFolder([string]$packagesFolder)
+    {
+        if(Test-Path $packagesFolder)
+        {
+            $packages = Get-ChildItem $packagesFolder\*.nupkg -Recurse
+            return $packages
+        }
+        return $null
+    }
+
+    function GetFiles([string]$folder)
+    {
+        if(Test-Path $folder)
+        {
+            $files = Get-ChildItem $folder -recurse
+            return $files
+        }
+        return $null
+    }
+
+    function GetNuGetFoldersPath()
+    {
+        $nugetFolder = [System.IO.Path]::Combine($env:UserProfile, "np")
+        return $nugetFolder
+    }
+
+    function IsClientDotnetExe([string]$nugetClient)
+    {
+        return $nugetClient.endswith("dotnet.exe")
+    }
+
+    function SetupNuGetFolders([string]$_nugetClientFilePath)
+    {
+        $nugetFolders = GetNuGetFoldersPath
+        $Env:NUGET_PACKAGES = [System.IO.Path]::Combine($nugetFolders, "gpf")
```

Comment:
same here, what's the reasoning behind the naming? should we add more descriptive names than `gpf`, `hcp`, `pcp`?

Reply:
gpf/hcp/pcp are just short names so that we don't hit any potential long path problems regardless of the project. 
so ~\np\gfp is the equivalent of ~\.nuget\packages but used for perf tests.. and likewise with the others.